tasks/update-granules-metadata/tests/test-index.js

```diff
@@ -0,0 +1,122 @@
+      ? parseS3Uri(file).Key : cmrReadStream(file),
+  })));
+}
+
+function granulesToFileURIs(granules) {
+  const s3URIs = granules.reduce((arr, g) => arr.concat(g.files.map((file) => file.filename)), []);
+  return s3URIs;
+}
+
+function buildPayload(t) {
+  const newPayload = t.context.payload;
+
+  newPayload.config.bucket = t.context.stagingBucket;
+  newPayload.config.buckets.internal.name = t.context.stagingBucket;
+  newPayload.config.buckets.public.name = t.context.publicBucket;
+  newPayload.config.buckets.protected.name = t.context.protectedBucket;
+
+  newPayload.input.granules.forEach((gran) => {
+    gran.files.forEach((file) => {
+      file.bucket = t.context.stagingBucket;
+      file.filename = buildS3Uri(t.context.stagingBucket, parseS3Uri(file.filename).Key);
+    });
+  });
+
+  return newPayload;
+}
+
+test.beforeEach(async (t) => {
+  t.context.stagingBucket = randomId('staging');
+  t.context.publicBucket = randomId('public');
+  t.context.protectedBucket = randomId('protected');
+  t.context.systemBucket = randomId('system');
+  t.context.stackName = randomString();
+  await Promise.all([
+    s3().createBucket({ Bucket: t.context.stagingBucket }).promise(),
+    s3().createBucket({ Bucket: t.context.publicBucket }).promise(),
+    s3().createBucket({ Bucket: t.context.protectedBucket }).promise(),
+    s3().createBucket({ Bucket: t.context.systemBucket }).promise(),
+  ]);
+  process.env.system_bucket = t.context.systemBucket;
+  process.env.stackName = t.context.stackName;
+  putJsonS3Object(
+    t.context.systemBucket,
+    getDistributionBucketMapKey(t.context.stackName),
+    {
+      [t.context.stagingBucket]: t.context.stagingBucket,
+      [t.context.publicBucket]: t.context.publicBucket,
+      [t.context.protectedBucket]: t.context.protectedBucket,
+      [t.context.systemBucket]: t.context.systemBucket,
+    }
+  );
+
+  const payloadPath = path.join(__dirname, 'data', 'payload.json');
+  const rawPayload = fs.readFileSync(payloadPath, 'utf8');
+  t.context.payload = JSON.parse(rawPayload);
+  const filesToUpload = granulesToFileURIs(t.context.payload.input.granules);
+  t.context.filesToUpload = filesToUpload.map((file) =>
+    buildS3Uri(`${t.context.stagingBucket}`, parseS3Uri(file).Key));
+  process.env.REINGEST_GRANULE = false;
+});
+
+test.afterEach.always(async (t) => {
+  await recursivelyDeleteS3Bucket(t.context.publicBucket);
+  await recursivelyDeleteS3Bucket(t.context.stagingBucket);
+  await recursivelyDeleteS3Bucket(t.context.protectedBucket);
+  await recursivelyDeleteS3Bucket(t.context.systemBucket);
+});
+
+test.serial('Should add etag to each CMR metadata file by checking that etag is one of more characters not whitespace', async (t) => {
+  const newPayload = buildPayload(t);
+  const filesToUpload = cloneDeep(t.context.filesToUpload);
+  await uploadFiles(filesToUpload, t.context.stagingBucket);
+
+  const output = await updateGranulesMetadata(newPayload);
+  output.granules[0].files
```

Comment:
Since your input payload has two granules, each with a CMR file, why not do this test against the CMR file for both granules?

Reply:
Good point, I forgot to add the second test!

Reply:
I would just update this one test to run the same assertion for the CMR file in both granules

Reply:
Yeah, that's what I ended up doing